apps/explorer/src/components/top-validators-card/TopValidatorsCard.tsx

```diff
@@ -50,91 +86,139 @@ export const STATE_DEFAULT: ValidatorState = {
+};
+
+export type SystemParams = {
+    type: '0x2::sui_system::SystemParameters';
+    fields: {
+        max_validator_candidate_count: number;
+        min_validator_stake: bigint;
+    };
+};
+
+export type ValidatorState = {
+    delegation_reward: number;
+    epoch: number;
+    id: { id: string; version: number };
+    parameters: SystemParams;
+    storage_fund: number;
+    treasury_cap: ObjFields;
+    validators: {
+        type: '0x2::validator_set::ValidatorSet';
+        fields: {
+            delegation_stake: bigint;
+            active_validators: Validator[];
+            next_epoch_validators: Validator[];
+            pending_removals: string;
+            pending_validators: string;
+            quorum_stake_threshold: bigint;
+            total_validator_stake: bigint;
+        };
+    };
+};
+
+export function sortValidatorsByStake(validators: Validator[]) {
+    validators.sort((a: Validator, b: Validator): number => {
+        if (a.fields.stake_amount < b.fields.stake_amount) return 1;
+        if (a.fields.stake_amount > b.fields.stake_amount) return -1;
+        return 0;
+    });
 }
 
-export function TopValidatorsCardAPI() {
-    const [showObjectState, setObjectState] = useState(STATE_DEFAULT);
-    const [loadState, setLoadState] = useState('pending');
-    const rpc = useRpc();
-    useEffect(() => {
-        getValidatorState(rpc)
-            .then((objState: ValidatorState) => {
-                setObjectState(objState);
-                setLoadState('loaded');
-            })
-            .catch((error: any) => {
-                console.log(error);
-                setLoadState('fail');
-            });
-    }, [rpc]);
-
-    if (loadState === 'loaded') {
-        return <TopValidatorsCard state={showObjectState as ValidatorState} />;
-    }
-    if (loadState === 'pending') {
-        return (
-            <div data-testid="validators-table">
-                <TabGroup>
-                    <TabList>
-                        <Tab>Validators</Tab>
-                    </TabList>
-                    <TabPanels>
-                        <TabPanel>
-                            <div title="Top Validators">
-                                <PlaceholderTable
-                                    rowCount={3}
-                                    rowHeight="13px"
-                                    colHeadings={[
-                                        'Name',
-                                        'Address',
-                                        'Pubkey Bytes',
-                                    ]}
-                                    colWidths={['135px', '220px', '220px']}
-                                />
-                            </div>
-                        </TabPanel>
-                    </TabPanels>
-                </TabGroup>
-            </div>
-        );
-    }
-    if (loadState === 'fail') {
-        return <ValidatorLoadFail />;
-    }
+function stakeColumn(validator: {
```

Comment:
This should just be turned into a component instead of a function.

Reply:
done